crates/sui-core/src/authority/consensus_tx_status_cache.rs

```diff
@@ -160,22 +191,52 @@ impl ConsensusTxStatusCache {
 
     pub async fn update_last_committed_leader_round(&self, leader_round: u32) {
         debug!("Updating last committed leader round: {}", leader_round);
+        // Send update through watch channel.
+        let _ = self.last_committed_leader_round_tx.send(Some(leader_round));
+
         let mut inner = self.inner.write();
+
+        // Remove transactions that are expired.
         while let Some((position, _)) = inner.transaction_status.first_key_value() {
             if position.block.round + CONSENSUS_STATUS_RETENTION_ROUNDS <= leader_round {
-                inner.transaction_status.pop_first();
+                let (pos, status) = inner.transaction_status.pop_first().unwrap();
+                // Ensure the transaction is not in the fastpath certified set.
+                if status == ConsensusTxStatus::FastpathCertified {
+                    assert!(inner.fastpath_certified.remove(&pos));
+                }
+            } else {
+                break;
+            }
+        }
+
+        // GC fastpath certified transactions.
+        // In theory, notify_read_transaction_status_change() could return `Rejected` status directly
+        // to waiters on GC'ed transactions.
+        // But it is necessary to track the number of fastpath certified status anyway for end of epoch.
+        // So rejecting every fastpath certified transaction here.
+        while let Some(position) = inner.fastpath_certified.first().cloned() {
+            if position.block.round + self.consensus_gc_depth <= leader_round {
+                // Reject GC'ed transactions that were previously fastpath certified.
+                self.set_transaction_status_inner(
```

Comment:
Why do you need to manually set reject status for these transactions here. Won't they get GC'd by consensus and then get set to status rejected in the normal path